backend-api/src/functions/asyncBiometricToken/asyncBiometricTokenHandler.test.ts

```diff
@@ -1,23 +1,71 @@
+import { Context } from "aws-lambda";
 import { Logger } from "../services/logging/logger";
 import { MockLoggingAdapter } from "../services/logging/tests/mockLogger";
 import { buildLambdaContext } from "../testUtils/mockContext";
 import { buildRequest } from "../testUtils/mockRequest";
 import { lambdaHandlerConstructor } from "./asyncBiometricTokenHandler";
-import { registeredLogs } from "./registeredLogs";
+import { IAsyncBiometricTokenDependencies } from "./handlerDependencies";
+import { MessageName, registeredLogs } from "./registeredLogs";
 
 describe("Async Biometric Token", () => {
-  describe("Given a request is made", () => {
+  let dependencies: IAsyncBiometricTokenDependencies;
+  let mockLoggingAdapter: MockLoggingAdapter<MessageName>;
+  let context: Context;
+
+  beforeEach(() => {
+    mockLoggingAdapter = new MockLoggingAdapter();
+    dependencies = {
+      logger: () => new Logger(mockLoggingAdapter, registeredLogs),
+    };
+    context = buildLambdaContext();
+  });
+  describe("Request body validation", () => {
+    describe("Given request body is invalid", () => {
+      it("Logs and returns 400 Bad Request response", async () => {
+        const request = buildRequest({
+          body: JSON.stringify({
+            sessionId: "58f4281d-d988-49ce-9586-6ef70a2be0b4",
+          }),
+        });
+
+        const result = await lambdaHandlerConstructor(
+          dependencies,
+          request,
+          context,
+        );
+
+        expect(mockLoggingAdapter.getLogMessages()[1].logMessage.message).toBe(
+          "REQUEST_BODY_INVALID",
+        );
+        expect(mockLoggingAdapter.getLogMessages()[1].data).toStrictEqual({
+          errorMessage:
+            "documentType in request body is either null or undefined. documentType: undefined",
+        });
+        expect(result).toStrictEqual({
+          headers: { "Content-Type": "application/json" },
+          statusCode: 400,
+          body: JSON.stringify({
+            error: "invalid_request",
+            error_description: "Request body invalid",
+          }),
+        });
+      });
+    });
+  });
+
+  describe("Given a valid request is made", () => {
     it("Logs and returns 501 Not Implemented response", async () => {
-      const mockLoggingAdapter = new MockLoggingAdapter();
-      const dependencies = {
-        logger: () => new Logger(mockLoggingAdapter, registeredLogs),
-      };
-      const event = buildRequest();
+      const request = buildRequest({
+        body: JSON.stringify({
+          sessionId: "58f4281d-d988-49ce-9586-6ef70a2be0b4",
```

Comment:
Nitpick: can we extract the session ID into a variable somewhere? Even though it's just an arbitrary string at the moment, it'll come up again in the handler logic and it'll be harder to maintain a bunch of identical strings, especially when they're this complex.

Reply:
Done